### ISSUE TITLE
False negative in typevar passed to generic function with not compatible type

### ISSUE THREAD

**Issue by minmax** [microsoft/pyright#7542]
**Describe the bug**
I thing call `cmp(a)` must be denied.

```python
import operator
from collections.abc import Callable
from collections.abc import Container
from functools import partial
from typing import TypeVar


T = TypeVar("T")
VT = TypeVar("VT")


def foo(a: T, b: Container[VT], cmp_with_error: Callable[[VT], bool]) -> T:
    cmp = contains(b)  # (VT@foo) -> bool
    cmp(a)  # False negative. T != VT
    cmp_with_error(a)  # ok, error detected: Type "T@foo" cannot be assigned to type "VT@foo"
    return a


def contains(container: Container[T]) -> Callable[[T], bool]:
    return partial(operator.contains, container)
```

pyright version: main (3da27fd5491f4e2e4d1e6965ac068de2acb062aa)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.358, which I just published.
